python310Packages.pylsp-mypy: build fails

### Steps To Reproduce
Steps to reproduce the behavior:
1. build `python310Packages.pylsp-mypy`

### Build log
```
❯ nix-build . -A python310Packages.pylsp-mypy 
this derivation will be built:
  /nix/store/mx1abx4nig7cjqyn010sdmkxxw7ka4ig-python3.10-pylsp-mypy-0.6.3.drv
building '/nix/store/mx1abx4nig7cjqyn010sdmkxxw7ka4ig-python3.10-pylsp-mypy-0.6.3.drv'...
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing setuptools-build-hook
Using setuptoolsBuildPhase
Using setuptoolsShellHook
Sourcing pip-install-hook
Using pipInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing setuptools-check-hook
Using setuptoolsCheckPhase
Sourcing pytest-check-hook
Using pytestCheckPhase
Removing setuptoolsCheckPhase
unpacking sources
unpacking source archive /nix/store/i96d92x36i34wgpv783x6pf73ik9i5gh-source
source root is source
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file source/test/test_plugin.py
patching sources
configuring
no configure script, doing nothing
building
Executing setuptoolsBuildPhase
running bdist_wheel
running build
running build_py
creating build
creating build/lib
creating build/lib/pylsp_mypy
copying pylsp_mypy/plugin.py -> build/lib/pylsp_mypy
copying pylsp_mypy/__init__.py -> build/lib/pylsp_mypy
copying pylsp_mypy/_version.py -> build/lib/pylsp_mypy
/nix/store/sl0bg0ariygr2cvpf0d7gd9jnclmi6ra-python3.10-setuptools-65.3.0/lib/python3.10/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
installing to build/bdist.linux-x86_64/wheel
running install
running install_lib
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/wheel
creating build/bdist.linux-x86_64/wheel/pylsp_mypy
copying build/lib/pylsp_mypy/plugin.py -> build/bdist.linux-x86_64/wheel/pylsp_mypy
copying build/lib/pylsp_mypy/__init__.py -> build/bdist.linux-x86_64/wheel/pylsp_mypy
copying build/lib/pylsp_mypy/_version.py -> build/bdist.linux-x86_64/wheel/pylsp_mypy
running install_egg_info
running egg_info
creating pylsp_mypy.egg-info
writing pylsp_mypy.egg-info/PKG-INFO
writing dependency_links to pylsp_mypy.egg-info/dependency_links.txt
writing entry points to pylsp_mypy.egg-info/entry_points.txt
writing requirements to pylsp_mypy.egg-info/requires.txt
writing top-level names to pylsp_mypy.egg-info/top_level.txt
writing manifest file 'pylsp_mypy.egg-info/SOURCES.txt'
reading manifest file 'pylsp_mypy.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
adding license file 'LICENSE'
writing manifest file 'pylsp_mypy.egg-info/SOURCES.txt'
Copying pylsp_mypy.egg-info to build/bdist.linux-x86_64/wheel/pylsp_mypy-0.6.3-py3.10.egg-info
running install_scripts
adding license file "LICENSE" (matched pattern "LICEN[CS]E*")
creating build/bdist.linux-x86_64/wheel/pylsp_mypy-0.6.3.dist-info/WHEEL
creating 'dist/pylsp_mypy-0.6.3-py3-none-any.whl' and adding 'build/bdist.linux-x86_64/wheel' to it
adding 'pylsp_mypy/__init__.py'
adding 'pylsp_mypy/_version.py'
adding 'pylsp_mypy/plugin.py'
adding 'pylsp_mypy-0.6.3.dist-info/LICENSE'
adding 'pylsp_mypy-0.6.3.dist-info/METADATA'
adding 'pylsp_mypy-0.6.3.dist-info/WHEEL'
adding 'pylsp_mypy-0.6.3.dist-info/entry_points.txt'
adding 'pylsp_mypy-0.6.3.dist-info/top_level.txt'
adding 'pylsp_mypy-0.6.3.dist-info/RECORD'
removing build/bdist.linux-x86_64/wheel
Finished executing setuptoolsBuildPhase
installing
Executing pipInstallPhase
/build/source/dist /build/source
Processing ./pylsp_mypy-0.6.3-py3-none-any.whl
Requirement already satisfied: python-lsp-server in /nix/store/r5iha88g6y77p30rrz5v984msr49jkar-python3.10-python-lsp-server-1.6.0/lib/python3.10/site-packages (from pylsp-mypy==0.6.3) (1.6.0)
Requirement already satisfied: mypy in /nix/store/nba9703m5ndw70m1aqa92zagv8afi6f4-python3.10-mypy-0.991/lib/python3.10/site-packages (from pylsp-mypy==0.6.3) (0.991)
Requirement already satisfied: toml in /nix/store/9f5b3fpdv7xrnncjicxhhy6wgmx7a1m5-python3.10-toml-0.10.2/lib/python3.10/site-packages (from pylsp-mypy==0.6.3) (0.10.2)
Requirement already satisfied: tomli>=1.1.0 in /nix/store/n3yvqabqnivjj149v31m1j2093g2i6rm-python3.10-tomli-2.0.1/lib/python3.10/site-packages (from mypy->pylsp-mypy==0.6.3) (2.0.1)
Requirement already satisfied: typing-extensions>=3.10 in /nix/store/j2nhmx4nbiqdg7nzbi9jxwjqym2q3fr3-python3.10-typing-extensions-4.3.0/lib/python3.10/site-packages (from mypy->pylsp-mypy==0.6.3) (4.3.0)
Requirement already satisfied: mypy-extensions>=0.4.3 in /nix/store/0lgg76zgknbb51b86jfzmbp7r8mpgy36-python3.10-mypy-extensions-0.4.3/lib/python3.10/site-packages (from mypy->pylsp-mypy==0.6.3) (0.4.3)
Requirement already satisfied: ujson>=3.0.0 in /nix/store/2nvih7avlw2yah33dxf5k2m1zhz8mb2v-python3.10-ujson-5.5.0/lib/python3.10/site-packages (from python-lsp-server->pylsp-mypy==0.6.3) (5.5.0)
Requirement already satisfied: jedi<0.19.0,>=0.17.2 in /nix/store/6yav0cwmkgyfhv0jf2ahy4vs6ifcz5xs-python3.10-jedi-0.18.1/lib/python3.10/site-packages (from python-lsp-server->pylsp-mypy==0.6.3) (0.18.1)
Requirement already satisfied: pluggy>=1.0.0 in /nix/store/mgjn879a43sjhlls2kq8454c2y6dsqx4-python3.10-pluggy-1.0.0/lib/python3.10/site-packages (from python-lsp-server->pylsp-mypy==0.6.3) (1.0.0)
Requirement already satisfied: docstring-to-markdown in /nix/store/hqfsv7vpzg7w1v49nf7n256fjyz2yfd4-python3.10-docstring-to-markdown-0.10/lib/python3.10/site-packages (from python-lsp-server->pylsp-mypy==0.6.3) (0.10)
Requirement already satisfied: python-lsp-jsonrpc>=1.0.0 in /nix/store/2xrz708914bjy7ab9179dhr7a136q4w7-python3.10-python-lsp-jsonrpc-1.0.0/lib/python3.10/site-packages (from python-lsp-server->pylsp-mypy==0.6.3) (1.0.0)
Requirement already satisfied: setuptools>=39.0.0 in /nix/store/sl0bg0ariygr2cvpf0d7gd9jnclmi6ra-python3.10-setuptools-65.3.0/lib/python3.10/site-packages (from python-lsp-server->pylsp-mypy==0.6.3) (65.3.0.post0)
Requirement already satisfied: parso<0.9.0,>=0.8.0 in /nix/store/lcpwqya3c9wpwmykdgd820h9jp3792fz-python3.10-parso-0.8.3/lib/python3.10/site-packages (from jedi<0.19.0,>=0.17.2->python-lsp-server->pylsp-mypy==0.6.3) (0.8.3)
Installing collected packages: pylsp-mypy
Successfully installed pylsp-mypy-0.6.3
/build/source
Finished executing pipInstallPhase
pythonOutputDistPhase
Executing pythonOutputDistPhase
Finished executing pythonOutputDistPhase
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/9f9vxj9mj4hpvakr368sh3kp566sq4ib-python3.10-pylsp-mypy-0.6.3
strip is /nix/store/dq0xwmsk1g0i2ayg6pb7y87na2knzylh-gcc-wrapper-11.3.0/bin/strip
stripping (with command strip and flags -S) in  /nix/store/9f9vxj9mj4hpvakr368sh3kp566sq4ib-python3.10-pylsp-mypy-0.6.3/lib
patching script interpreter paths in /nix/store/9f9vxj9mj4hpvakr368sh3kp566sq4ib-python3.10-pylsp-mypy-0.6.3
checking for references to /build/ in /nix/store/9f9vxj9mj4hpvakr368sh3kp566sq4ib-python3.10-pylsp-mypy-0.6.3...
shrinking RPATHs of ELF executables and libraries in /nix/store/v270g8cs6fwnbgpk2vzsf12k5dr8f2fh-python3.10-pylsp-mypy-0.6.3-dist
strip is /nix/store/dq0xwmsk1g0i2ayg6pb7y87na2knzylh-gcc-wrapper-11.3.0/bin/strip
patching script interpreter paths in /nix/store/v270g8cs6fwnbgpk2vzsf12k5dr8f2fh-python3.10-pylsp-mypy-0.6.3-dist
checking for references to /build/ in /nix/store/v270g8cs6fwnbgpk2vzsf12k5dr8f2fh-python3.10-pylsp-mypy-0.6.3-dist...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
running install tests
no Makefile or custom installCheckPhase, doing nothing
pythonCatchConflictsPhase
pythonRemoveBinBytecodePhase
pythonImportsCheckPhase
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: pylsp_mypy
pytestCheckPhase
Executing pytestCheckPhase
============================= test session starts ==============================
platform linux -- Python 3.10.8, pytest-7.1.3, pluggy-1.0.0
rootdir: /build/source
collected 11 items / 1 deselected / 10 selected                                

test/test_plugin.py .F.....F..                                           [100%]

=================================== FAILURES ===================================
_________________________________ test_plugin __________________________________

workspace = <pylsp.workspace.Workspace object at 0x7ffff58c62c0>
last_diagnostics_monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7ffff40e5450>

    def test_plugin(workspace, last_diagnostics_monkeypatch):
        config = FakeConfig()
        doc = Document(DOC_URI, workspace, DOC_TYPE_ERR)
        plugin.pylsp_settings(config)
        diags = plugin.pylsp_lint(config, workspace, doc, is_saved=False)
    
        assert len(diags) == 1
        diag = diags[0]
>       assert diag["message"] == TYPE_ERR_MSG
E       assert '"Dict[<nothi...attr-defined]' == '"Dict[<nothi...bute "append"'
E         Skipping 43 identical leading characters in diff, use -v to show
E         - te "append"
E         + te "append"  [attr-defined]

test/test_plugin.py:66: AssertionError
___________________________ test_multiple_workspaces ___________________________

tmpdir = local('/build/pytest-of-nixbld/pytest-0/test_multiple_workspaces0')
last_diagnostics_monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7ffff2027850>

    def test_multiple_workspaces(tmpdir, last_diagnostics_monkeypatch):
        DOC_SOURCE = """
    def foo():
        return
        unreachable = 1
    """
        DOC_ERR_MSG = "Statement is unreachable"
    
        # Initialize two workspace folders.
        folder1 = tmpdir.mkdir("folder1")
        ws1 = Workspace(uris.from_fs_path(str(folder1)), Mock())
        ws1._config = Config(ws1.root_uri, {}, 0, {})
        folder2 = tmpdir.mkdir("folder2")
        ws2 = Workspace(uris.from_fs_path(str(folder2)), Mock())
        ws2._config = Config(ws2.root_uri, {}, 0, {})
    
        # Create configuration file for workspace folder 1.
        mypy_config = folder1.join("mypy.ini")
        mypy_config.write("[mypy]\nwarn_unreachable = True\ncheck_untyped_defs = True")
    
        # Initialize settings for both folders.
        plugin.pylsp_settings(ws1._config)
        plugin.pylsp_settings(ws2._config)
    
        # Test document in workspace 1 (uses mypy.ini configuration).
        doc1 = Document(DOC_URI, ws1, DOC_SOURCE)
        diags = plugin.pylsp_lint(ws1._config, ws1, doc1, is_saved=False)
        assert len(diags) == 1
        diag = diags[0]
>       assert diag["message"] == DOC_ERR_MSG
E       AssertionError: assert 'Statement is...[unreachable]' == 'Statement is unreachable'
E         - Statement is unreachable
E         + Statement is unreachable  [unreachable]
E         ?                         +++++++++++++++

test/test_plugin.py:134: AssertionError
=========================== short test summary info ============================
FAILED test/test_plugin.py::test_plugin - assert '"Dict[<nothi...attr-defined...
FAILED test/test_plugin.py::test_multiple_workspaces - AssertionError: assert...
================== 2 failed, 8 passed, 1 deselected in 2.71s ===================
error: builder for '/nix/store/mx1abx4nig7cjqyn010sdmkxxw7ka4ig-python3.10-pylsp-mypy-0.6.3.drv' failed with exit code 1;
       last 10 log lines:
       > E       AssertionError: assert 'Statement is...[unreachable]' == 'Statement is unreachable'
       > E         - Statement is unreachable
       > E         + Statement is unreachable  [unreachable]
       > E         ?                         +++++++++++++++
       >
       > test/test_plugin.py:134: AssertionError
       > =========================== short test summary info ============================
       > FAILED test/test_plugin.py::test_plugin - assert '"Dict[<nothi...attr-defined...
       > FAILED test/test_plugin.py::test_multiple_workspaces - AssertionError: assert...
       > ================== 2 failed, 8 passed, 1 deselected in 2.71s ===================
       For full logs, run 'nix log /nix/store/mx1abx4nig7cjqyn010sdmkxxw7ka4ig-python3.10-pylsp-mypy-0.6.3.drv'.
```

### Notify maintainers
@cpcloud 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.79, NixOS, 23.05 (Stoat), 23.05.20221125.5dc7114`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.0`
 - channels(gaetan): `""`
 - channels(root): `"nixos"`
 - nixpkgs: `/etc/nix/inputs/nixpkgs`
```

--- a/pkgs/development/python-modules/pylsp-mypy/default.nix
+++ b/pkgs/development/python-modules/pylsp-mypy/default.nix
@@ -1,6 +1,7 @@
 { lib
 , buildPythonPackage
 , fetchFromGitHub
+, fetchpatch
 , mypy
 , pytestCheckHook
 , python-lsp-server
@@ -21,6 +22,14 @@ buildPythonPackage rec {
     rev = "refs/tags/${version}";
     hash = "sha256-fZ2bPPjBK/H2jMI4S3EhvWJaNl4tK7HstxcHSAkoFW4=";
   };
+
+  patches = [
+    (fetchpatch {
+      name = "0001-adapt-test-to-latest-mypy.patch";
+      url = "https://github.com/python-lsp/pylsp-mypy/commit/99cf687798464f810b128881dbbec82aa5353e04.patch";
+      sha256 = "sha256-wLaGMaW/gTab2fX7zGnemLQQNDWxBURYb7VsgEas61Y=";
+    })
+  ];
 
   propagatedBuildInputs = [
     mypy

